nupicaudio/encoders/iteration.py

```diff
@@ -0,0 +1,65 @@
+from nupic.encoders.base import Encoder
+import abc
+
+class IterationFinished(Exception):
+  """ 
+  EOF exception
+  """
+  def __init__(self, msg):
+    super(IterationFinished, self).__init__(msg)
+
+
+
+class IterationEncoder(Encoder):
+  __metaclass__ = abc.ABCMeta
+  """ 
+  Iteration encoder encapsulates parsing of some data, and then creates an iterator over the
+  (otherwise monolithic) file. 
+
+  An example is a WAV file, during init() the file is transformed to a numeric representation (array), 
+  and each call of encode() moves over one element of this array. 
+
+  Typical applicatios for IterationEncoder: binary files (sound, video, biosignals, ...)
```

Comment:
this is not a 100% pure "encoder" functionality, but I like how it works for the binary files, esp. WAV,...